core/runtime/jsruntime.rs

```diff
@@ -1961,9 +1978,21 @@ impl JsRuntime {
     scope: &mut v8::HandleScope,
     context_state: &RefCell<ContextState>,
   ) -> Result<bool, Error> {
-    let mut context_state = context_state.borrow_mut();
     let mut dispatched_ops = false;
 
+    // Poll any pending task spawner tasks. Note that we need to poll separately because otherwise
+    // Rust will extend the lifetime of the borrow longer than we expect.
+    let tasks = context_state.borrow().task_spawner_factory.poll_inner(cx);
```

Comment:
I assume this is free if there are no tasks?

Reply:
Correct. Just one boolean check (atomic but cheap). Will also go away with wakers